src/plugins/data/public/ui/query_editor/no_data_popover.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactElement, useEffect, useState } from 'react';
+import React from 'react';
+import { EuiButtonEmpty, EuiText, EuiTourStep } from '@elastic/eui';
+import { IStorageWrapper } from 'src/plugins/opensearch_dashboards_utils/public';
+import { i18n } from '@osd/i18n';
+
+const NO_DATA_POPOVER_STORAGE_KEY = 'data.noDataPopover';
+
+export function NoDataPopover({
+  showNoDataPopover,
+  storage,
+  children,
+}: {
+  showNoDataPopover?: boolean;
+  storage: IStorageWrapper;
+  children: ReactElement;
+}) {
+  const [noDataPopoverDismissed, setNoDataPopoverDismissed] = useState(() =>
+    Boolean(storage.get(NO_DATA_POPOVER_STORAGE_KEY))
+  );
+  const [noDataPopoverVisible, setNoDataPopoverVisible] = useState(false);
+
+  useEffect(() => {
+    if (showNoDataPopover && !noDataPopoverDismissed) {
+      setNoDataPopoverVisible(true);
+    }
+  }, [noDataPopoverDismissed, showNoDataPopover]);
+
+  return (
```

Comment:
I see we are repeating query_string_input 's context in query_editor. I know that there are there are three cases that we use query string input and only one with both settings enabled to use query editor, so if that is the case can't we just simply reuse these repeat component from query_string_input? why duplicate these component?

Reply:
yeah good catch. part of this issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6957. 

the goal here was to release to this for 2.15, it would keep the query string input completely intact (even restored some of the files to their pre-2.14 state). to ensure the code paths are completely separate in case users toggle it on and off. 

then targeting 2.16, with more automated testing start cleaning up these components.